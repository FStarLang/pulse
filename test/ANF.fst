module ANF

#lang-pulse
open Pulse

fn test ()
{
  let mut r = 0;
  r := !r + 1;
  r := !r + 1;
  r := !r + 1;
  r := !r + 1;
}
