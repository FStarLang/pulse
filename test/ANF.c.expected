/* krml header omitted for test repeatability */


#include "ANF.h"

extern krml_checked_int_t Prims_op_Addition(krml_checked_int_t x, krml_checked_int_t y);

void ANF_test(void)
{
  krml_checked_int_t r = (krml_checked_int_t)0;
  krml_checked_int_t __anf0 = r;
  r = Prims_op_Addition(__anf0, (krml_checked_int_t)1);
  krml_checked_int_t __anf01 = r;
  r = Prims_op_Addition(__anf01, (krml_checked_int_t)1);
  krml_checked_int_t __anf02 = r;
  r = Prims_op_Addition(__anf02, (krml_checked_int_t)1);
  krml_checked_int_t __anf03 = r;
  r = Prims_op_Addition(__anf03, (krml_checked_int_t)1);
}

